### PR TITLE
Len zero

### PR DESCRIPTION
**Pull request by llogiq**
I finally got the method lookup to run, now we no longer report the false positive on an `impl` without `is_empty()` method.

### CODE REVIEW

**Comment by llogiq** [comment]
I'll have to merge the current version first.

File: src/len_zero.rs [hunk 1]
```diff
@@ -92,10 +98,30 @@ fn check_cmp(cx: &Context, span: Span, left: &Expr, right: &Expr, empty: &str) {
...
+	fn check_item(cx: &Context, id: &ImplOrTraitItemId) -> bool {
+		if let &MethodTraitItemId(ref def_id) = id {
+			if let ty::MethodTraitItem(ref method) = ty::impl_or_trait_item(
+					cx.tcx, *def_id) {
+				method.name.as_str() == "is_empty"
+			} else { false }
+		} else { false }
+	}
+	
+	::rustc::middle::ty::ty_to_def_id(ty).map_or(true, |id| {
```
Comment (Manishearth) [inline src/len_zero.rs]: nit: `ty` is already imported, just use `ty::`

File: src/len_zero.rs [hunk 2]
```diff
@@ -92,10 +98,30 @@ fn check_cmp(cx: &Context, span: Span, left: &Expr, right: &Expr, empty: &str) {
...
+		cx.tcx.impl_items.borrow().get(&id).map_or(false, |item_ids| {
```
Comment (Manishearth) [inline src/len_zero.rs]: nit: some comments please :)

File: tests/run-pass.rs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#![feature(plugin)]
```
Comment (Manishearth) [inline tests/run-pass.rs]: nit: don't call this run-pass.rs, call it `mut_mut_macro.rs`

**Comment by llogiq** [comment]
I'm not sure what keeps git from merging it? @Manishearth can you merge it?

**Comment by Manishearth** [comment]
There's a merge conflict, you need to rebase and fix the errors.

**Pull request merged**